application/src/main/java/org/togetherjava/tjbot/commands/moderation/scam/ScamBlocker.java

```diff
@@ -0,0 +1,356 @@
+    }
+
+    private void reportScamMessage(@NotNull GuildMessageReceivedEvent event,
+            @NotNull String reportTitle, @Nullable ActionRow confirmDialog) {
+        Guild guild = event.getGuild();
+        Optional<TextChannel> reportChannel = getReportChannel(guild);
+        if (reportChannel.isEmpty()) {
+            logger.warn(
+                    "Unable to report a scam message, did not find a report channel matching the configured pattern '{}' for guild '{}'",
+                    reportChannelPattern, guild.getName());
+            return;
+        }
+
+        User author = event.getAuthor();
+        MessageEmbed embed =
+                new EmbedBuilder().setDescription(event.getMessage().getContentStripped())
+                    .setTitle(reportTitle)
+                    .setAuthor(author.getAsTag(), null, author.getAvatarUrl())
+                    .setTimestamp(event.getMessage().getTimeCreated())
+                    .setColor(AMBIENT_COLOR)
+                    .setFooter(author.getId())
+                    .build();
+        Message message =
+                new MessageBuilder().setEmbeds(embed).setActionRows(confirmDialog).build();
+
+        reportChannel.orElseThrow().sendMessage(message).queue();
+    }
+
+    private void dmUser(@NotNull GuildMessageReceivedEvent event) {
+        dmUser(event.getGuild(), event.getAuthor().getIdLong(), event.getJDA());
+    }
+
+    private void dmUser(@NotNull Guild guild, long userId, @NotNull JDA jda) {
+        String dmMessage =
+                """
+                        Hey there, we detected that you did send scam in the server %s and therefore put you under quarantine.
+                        This means you can no longer interact with anyone in the server until you have been unquarantined again.
+
+                        If you think this was a mistake (for example, your account was hacked, but you got back control over it),
+                        please contact a moderator or admin of the server.
+                        """
+                    .formatted(guild.getName());
+
+        jda.openPrivateChannelById(userId)
+            .flatMap(channel -> channel.sendMessage(dmMessage))
+            .queue();
+    }
+
+    private @NotNull Optional<TextChannel> getReportChannel(@NotNull Guild guild) {
+        return guild.getTextChannelCache().stream().filter(isReportChannel).findAny();
+    }
+
+    private @NotNull ActionRow createConfirmDialog(@NotNull GuildMessageReceivedEvent event) {
+        ComponentIdArguments args = new ComponentIdArguments(mode, event.getGuild().getIdLong(),
+                event.getChannel().getIdLong(), event.getMessageIdLong(),
+                event.getAuthor().getIdLong(),
+                ScamHistoryStore.hashMessageContent(event.getMessage()));
+
+        return ActionRow.of(Button.of(ButtonStyle.SUCCESS, generateComponentId(args), "Yes"),
+                Button.of(ButtonStyle.DANGER, generateComponentId(args), "No"));
```

Comment:
```suggestion
        return ActionRow.of(Button.success(generateComponentId(args), "Yes"),
                Button.danger(generateComponentId(args), "No"));
```

Reply:
cool

Reply:
You & your personality yes

Reply:
this is actually all over in the project. nice

Reply:
fixed

Reply:
Keep forgetting it myself too sssh